pkg/engine2/solution_context/resource_configuration.go

```diff
@@ -84,6 +112,33 @@ func ConstraintOperatorToAction(op constraints.ConstraintOperator) (string, erro
 	}
 }
 
+func getResourcesFromValue(val any) (ids []construct.ResourceId) {
+	if val == nil {
+		return
+	}
+	switch v := val.(type) {
+	case construct.ResourceId:
+		ids = []construct.ResourceId{v}
+	case construct.PropertyRef:
+		ids = []construct.ResourceId{v.Resource}
+	default:
+		rval := reflect.ValueOf(val)
+		switch rval.Kind() {
+		case reflect.Slice, reflect.Array:
+			for i := 0; i < reflect.ValueOf(val).Len(); i++ {
+				idVal := rval.Index(i).Interface()
+				ids = append(ids, getResourcesFromValue(idVal)...)
+			}
+		case reflect.Map:
+			for _, key := range reflect.ValueOf(val).MapKeys() {
```

Comment:
What about keys, could there ever be something like a `map(resource, X)`?

Reply:
all of our properties are string keys right?

Reply:
If that's the case, then the type should be `map(valueType)` and not let it be configurable to make sure.